src/chop/nn/snn/auto_cuda/cfunction.py

```diff
@@ -0,0 +1,418 @@
+from typing import Optional
```

Comment:
same as `auto_cuda/base.py`. please leave some references if possible like the link to the source code.